x-pack/legacy/plugins/monitoring/server/lib/alerts/license_expiration.lib.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import moment, { Moment } from 'moment-timezone';
+import { i18n } from '@kbn/i18n';
+import { AlertInstance } from '../../../../alerting/server/alert_instance';
+import { AlertLicense } from '../../alerts/types';
+// @ts-ignore
+import { formatTimestampToDuration } from '../../../common';
+import { CALCULATE_DURATION_UNTIL } from '../../../common/constants';
+
+const RESOLVED_SUBJECT = i18n.translate(
+  'xpack.monitoring.alerts.licenseExpiration.resolvedSubject',
+  {
+    defaultMessage: 'RESOLVED X-Pack Monitoring: License Expiration',
+  }
+);
+
+const NEW_SUBJECT = i18n.translate('xpack.monitoring.alerts.licenseExpiration.newSubject', {
+  defaultMessage: 'NEW X-Pack Monitoring: License Expiration',
+});
+
+export function executeActions(
+  instance: AlertInstance,
+  license: AlertLicense,
+  $expiry: Moment,
+  dateFormat: string,
+  emailAddress: string,
+  resolved: boolean = false
+) {
+  if (resolved) {
+    instance.scheduleActions('default', {
+      subject: RESOLVED_SUBJECT,
+      message: `This cluster alert has been resolved: Cluster '${
+        license.clusterName
+      }' license was going to expire on ${$expiry.format(dateFormat)}.`,
+      to: emailAddress,
+    });
+  } else {
+    instance.scheduleActions('default', {
+      subject: NEW_SUBJECT,
+      message: `Cluster '${license.clusterName}' license is going to expire on ${$expiry.format(
+        dateFormat
+      )}. Please update your license.`,
+      to: emailAddress,
+    });
+  }
+}
+
+export function getUiMessage(license: AlertLicense, timezone: string, resolved: boolean = false) {
+  if (resolved) {
+    return i18n.translate('xpack.monitoring.alerts.licenseExpiration.ui.resolvedMessage', {
+      defaultMessage: `This cluster's license is active.`,
+    });
+  }
+  return i18n.translate('xpack.monitoring.alerts.licenseExpiration.ui.firingMessage', {
+    defaultMessage: `This cluster's license is going to expire in {relative} at {absolute}.`,
+    values: {
+      relative: formatTimestampToDuration(license.expiryDateMS, CALCULATE_DURATION_UNTIL, null),
+      absolute: moment
+        .tz(license.expiryDateMS, timezone !== 'Browser' ? timezone : moment.tz.guess())
```

Comment:
This code will execute on the server (and not in the browser), thus always returning server tz (most likely utc). I think this should return utc time and we can do tz offsetting in the browser

Reply:
Yea this is a good point. We really need to be doing this on the client. I'll fix it up